UnityProject/Assets/Scripts/Communications/SignalReceiver.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Managers;
+using Mirror;
+using UnityEngine;
+using ScriptableObjects.Communications;
+
+namespace Communications
+{
+	public abstract class SignalReceiver : NetworkBehaviour
+	{
+		public SignalType SignalTypeToReceive = SignalType.PING;
+		public float Frequency = 122F;
+		public SignalEmitter Emitter;
+		public float DelayTime = 3f; //How many seconds of delay before the SignalReceive logic happens for weak signals
+
+
+		private void OnEnable()
+		{
+			if(CustomNetworkManager.Instance._isServer == false) return;
+			SignalsManager.Instance.Receivers.Add(this);
+		}
+
+		private void OnDisable()
+		{
+			if(CustomNetworkManager.Instance._isServer == false) return;
+			SignalsManager.Instance.Receivers.Remove(this);
+		}
+
+		/// <summary>
+		/// Logic to do when
+		/// </summary>
+		public abstract void ReceiveSignal(SignalStrength strength);
```

Comment:
maybe this could be called ``OnSignalReceived``

Reply:
Signal Manager uses ReceiveSignal so calling it "OnSignalReceive" won't make sense.

Reply:
![image](https://user-images.githubusercontent.com/34368774/140187404-80aa9560-8cfe-4fe9-9ea2-bc8441244b10.png)

Reply:
it sounds better in my head but I am not making it a demand, just a suggestion. You can resolve this conversation if you don't  feel like changing it.